packages/api/src/chat/flow-generation-parameters-schema.ts

```diff
@@ -23,27 +23,72 @@ export type FlowGenerationParameters = {
   name: string;
   description: string;
   prompt: string;
+  parameters?: Array<FlowParameter>;
 };
 
-export const FlowGenerationParametersSchema: ZodType<FlowGenerationParameters> = z.object({
-  name: z
-    .string()
-    .describe(
-      `A unique name for the flow, formatted as a DNS subdomain (e.g., "my-new-flow"). It must be lowercase and contain only letters, numbers, and hyphens. It cannot start or end with a hyphen.`,
-    )
-    .transform(val =>
-      val
-        .trim()
-        .toLowerCase()
-        .replace(/\s+/g, '-') // Replace spaces with hyphens
-        .replace(/[^a-z0-9-]/g, '') // Remove all other invalid characters
-        .slice(0, 63),
-    )
-    .pipe(z.string().regex(/^[a-z0-9]([-a-z0-9]{0,61}[a-z0-9])?$/, 'Invalid DNS subdomain name after transformation.')),
-  description: z.string().describe('Description of the flow, give a short description of what the flow does.'),
-  prompt: z
-    .string()
-    .describe(
-      'Help me create a reproducible prompt that achieves the same result as in the conversation above. The prompt will be executed by another LLM without any further user input, so it must include all the necessary information to reproduce the same outcome.',
-    ),
+export type FlowParameterAIGenerated = {
+  name: string;
+  description: string;
+  format: string;
+  default?: string;
+};
+
+export type FlowParameter = FlowParameterAIGenerated & {
+  required: boolean;
+};
+
+type FlowAIGeneratedParameters = {
+  name: string;
+  description: string;
+  prompt: string;
+  parameters?: Array<FlowParameterAIGenerated>;
+};
+
+export const FlowParameterSchema = z.object({
+  name: z.string().describe('Parameter name (must be valid identifier)'),
+  format: z.string().default('string').describe('Parameter data type'),
+  description: z.string().describe('Human-readable description of the parameter'),
+  default: z.string().optional().describe('Default value for the parameter'),
+  // Note: 'required' is computed dynamically based on whether 'default' exists
+});
+
+// Reusable schema fields for flow metadata
+const flowNameSchema = z
```

Comment:
issue(blocking): please introduce this change in a separate commit or PR because you change the existing schema construction before adding the parameters field.